.github/workflows/test_pull_requests.yml

```diff
@@ -0,0 +1,199 @@
+        echo MACOSX_DEPLOYMENT_TARGET=${{ matrix.macos_dev_target }} >> $GITHUB_ENV
+
+    - name: Install Dependencies for Linux
+      if: runner.os == 'Linux'
+      run: |
+        sudo apt-get update
+        sudo apt-get install libxkbcommon-x11-0 xorg-dev libgl1-mesa-dev
+        if [[ "${{ matrix.os }}" == "ubuntu-24.04" ]]; then
+          # https://github.com/actions/runner-images/issues/10025
+          echo "FC=gfortran-13" >> $GITHUB_ENV
+        fi
+
+    # BUILD AND TEST INTEGRATION FILES ON THE BASELINE BRANCH
+
+    - name: Baseline Checkout
+      if: matrix.run_regressions
+      uses: actions/checkout@v4
+      with:
+        ref: develop
+        path: baseline
+
+    - name: Baseline Create Build Directory
+      if: matrix.run_regressions
+      run: cmake -E make_directory ./baseline/build/
+
+    - name: Baseline Configure CMake
+      if: matrix.run_regressions
+      working-directory: ./baseline/build
+      run: >
+        cmake 
+        -G "${{ matrix.generator }}"
+        -DCMAKE_BUILD_TYPE:STRING=Release
+        -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=${{ matrix.macos_dev_target }}
+        -DLINK_WITH_PYTHON:BOOL=ON
+        -DPython_REQUIRED_VERSION:STRING=${{ steps.setup-python.outputs.python-version }}
+        -DPython_ROOT_DIR:PATH=$RUNNER_TOOL_CACHE/Python/${{ steps.setup-python.outputs.python-version }}/${{ matrix.python-arch }}/
+        -DBUILD_TESTING:BOOL=ON
+        -DBUILD_FORTRAN:BOOL=ON 
+        -DBUILD_PACKAGE:BOOL=OFF
+        -DDOCUMENTATION_BUILD:STRING=DoNotBuild
+        -DENABLE_OPENMP:BOOL=OFF
+        -DUSE_OpenMP:BOOL=OFF
+        ../
+
+    # During baseline builds, just build specific target list so that we don't waste time building the unit test binary
+
+    - name: Baseline Build on Windows
+      if: matrix.run_regressions && runner.os == 'Windows'
+      working-directory: ./baseline/build
+      run: cmake --build . -j ${{ matrix.nproc }} --config Release --target energyplus ExpandObjects_build ReadVars_build Slab_build Basement_build AppGPostProcess_build ParametricPreprocessor_build
+
+    - name: Baseline Build on Mac/Linux
+      if: matrix.run_regressions && runner.os != 'Windows'
+      working-directory: ./baseline/build
+      run: cmake --build . -j ${{ matrix.nproc }} --target energyplus ExpandObjects ReadVarsESO Slab Basement AppGPostProcess ParametricPreprocessor
+
+    - name: Baseline Test
+      if: matrix.run_regressions
+      working-directory: ./baseline/build
+      run: ctest -C Release -E Basement -R integration -j 3  # TODO: Speed up basement so we don't have to skip it.
+
+    # BUILD AND TEST EVERYTHING ON THE CURRENT BRANCH
+
+    - name: Branch Checkout
+      uses: actions/checkout@v4
+      with:
+        path: branch
+
+    - name: Branch Create Build Directory
+      run: cmake -E make_directory ./branch/build/
+
+    - name: Branch Configure CMake
+      working-directory: ./branch/build
+      run: >
```

Comment:
I was hesitating on this because I didn't want to add more complication while figuring all this out.  But if using ccache is straightforward, then by all means we should enable it.  I'll look into it.